roles/ovirt-cluster-upgrade/tasks/pinned_vms.yml

```diff
@@ -0,0 +1,25 @@
+- name: Create list of host IDs which has pinned VM and are UP
+  set_fact:
+    host_ids_items: "{{ item.placement_policy.hosts[0].id }}"
+  with_items:
+    - "{{ ovirt_vms | default([]) }}"
+  when: "{{ 'hosts' in item['placement_policy'] and item.placement_policy.hosts | length == 1 and item.status == 'up' }}"
+  register: host_ids_result
+
+- name: Create list of host IDs which has pinned VM
+  set_fact:
+    host_ids: "{{ host_ids_result.results | rejectattr('ansible_facts', 'undefined') | map(attribute='ansible_facts.host_ids_items') | list }}"
+
+- block:
+  - name: Shutdown pin to one host VMs
```

Comment:
It is optional, if user set `force_pinned=false` (the default) we simply skip it.